src/Api/Persistent.hs

```diff
@@ -159,6 +160,14 @@ getSpaces = getDb dbSpaces
 getIdeas :: Persist [Idea]
 getIdeas = getDb dbIdeas
 
+-- | If idea space already exists, return 'False'.  Otherwise, create it and return 'True'.
+addIdeaSpace :: IdeaSpace -> Persist Bool
```

Comment:
but if it throws an error, it won't hide information, it will just be more complicated to call it if you don't care whether it's there or not.

I don't think hiding the information whether an idea space exists is interesting from a security perspective.

I'm just thinking that I want to rename it to `addIdeaSpaceIfNotExists`.